packages/cli-doctor/src/tools/healthchecks/xcodeEnv.ts

```diff
@@ -0,0 +1,63 @@
+import {HealthCheckInterface} from '../../types';
+import fs from 'fs';
+import {promisify} from 'util';
+import {findProjectRoot} from '@react-native-community/cli-tools';
+
+const applePlatformFolders = ['ios', 'macos'];
+const xcodeEnvFile = '.xcode.env';
+
+function allAppleProjectsHaveXcodeEnv(projectRoot: string): boolean {
+  return applePlatformFolders
+    .map((folder) => xcodeEnvFileExistsFor(folder, projectRoot) ?? true)
+    .reduce((previousValue, currentValue) => previousValue && currentValue);
+}
+
+/**
+ * Checks whether the basePath+platform folder exists. If that happens, it means
+ * that the project supports that platform.
+ * Returns whether the file exists in a supported platform. It returns `null` if
+ * the platform is not supported
+ * @param platform the platform to be checked
+ * @param basePath the path to the parent of the platform
+ * @returns If the platform is supported, `true` if there is the file, `false` otherwise. If the platform is not supported, `null`
+ */
+function xcodeEnvFileExistsFor(
+  platform: string,
+  basePath: string,
+): boolean | null {
+  const platformPath = basePath + platform;
+  if (fs.existsSync(platformPath)) {
+    return fs.existsSync(platformPath + xcodeEnvFile);
+  }
+  return null;
+}
+
+export default {
+  label: '.xcode.env',
+  description: 'File to customize Xcode environment',
+  getDiagnostics: async () => {
+    const projectRoot = findProjectRoot();
+    const allExists = allAppleProjectsHaveXcodeEnv(projectRoot);
+    return {
+      needsToBeFixed: !allExists,
+    };
+  },
+  runAutomaticFix: async () => {
+    const templateFilePath = '_xcode.env';
+    const projectRoot = findProjectRoot();
+
+    const src =
+      projectRoot +
+      '/node_modules/react-native/template/ios/' +
+      templateFilePath;
+    const copyFileAsync = promisify(fs.copyFile);
+
+    applePlatformFolders.forEach(async (folder) => {
+      const destFolder = projectRoot + folder;
+      const destFilePath = destFolder + '/' + xcodeEnvFile;
+      if (fs.existsSync(destFolder) && !fs.existsSync(destFilePath)) {
+        await copyFileAsync(src, destFilePath);
+      }
+    });
```

Comment:
This is assuming the location of the `.xcworkspace`. I've seen many differing setups including:

- `<project-root>/ios/MyApp.xcworkspace`
- `<project-root>/MyApp.xcworkspace`
- `<project-root>/example/ios/MyApp.xcworkspace`
- `<project-root>/apps/ios/src/MyApp.xcworkspace`

Can we refactor `findPodfilePath` to return all found `Podfile`s and add a `.xcode.env` next to them? I think that will be more accurate.

Reply:
Or introduce a `findAllPodfilePaths` helper

Reply:
Hi! Thank you all for the suggestions and @tido64 for sharing its experience.
I updated the PR with the `findAllPodfilePaths` function, as suggested by @thymikee and adding the file in all the returned paths.